clang/test/SemaCXX/warn-unused-private-field.cpp

```diff
@@ -20,6 +20,17 @@ class SpaceShipDefaultCompare {
   int operator<=>(const SpaceShipDefaultCompare &) const = default;
 };
 
+class UnusedConstPrivateField {
+ public:
+  UnusedConstPrivateField() : unused_(0) {}
+ private:
+  const int unused_; // expected-warning{{private field 'unused_' is not used}}
+};
+
+class FriendEqDefaultCompare {
+  friend auto operator==(FriendEqDefaultCompare, FriendEqDefaultCompare) -> bool = default;
```

Comment:
Can you add a test case with a non-friend (normal method) on the unrelated class too?
Also it would be nice to extend the original test (the positive one) with a case where the comparison is defined as a friend function.

Reply:
I've updated the tests, hopefully I didn't misinterpret your comments. I added a non-friend method to the unrelated class and added a case where the comparison is defined as a friend function.